### PR TITLE
Cherry-picking charger led and SOC commit onto develop/shepherd-fw

### PR DESCRIPTION
**Pull request by nwdepatie**
## Issue ticket number and issue description

Ticket #99   -> Add charger LED and calculation of state of charge of pack

## Describe your changes

Basically the same as [this PR](https://github.com/Northeastern-Electric-Racing/shepherd_bms/pull/118), but I just cleaned up the commits because it had hanging commits from the ```develop/weekend-fix``` branch that was already merged in

### CODE REVIEW

File: shepherd_bms/include/analyzer.h [hunk 1]
```diff
@@ -80,6 +80,18 @@ class Analyzer
             20, 15, 10, 5, 1, 1
         };
 
+        /**
+         * @brief Lookup table for State of Charge
+         * 
+         * @note Each index corresponds to 10% charge
```
Comment (mccauleyma) [inline shepherd_bms/include/analyzer.h:86]: ?? This comment doesn't make sense to me

File: shepherd_bms/src/analyzer.cpp [hunk 2]
```diff
@@ -384,4 +385,18 @@ void Analyzer::disableTherms()
             }
         }
     }
+}
+
+void Analyzer::calcStateOfCharge()
+{
+    int index = (((bmsdata->minVoltage.val) - MIN_VOLT) / .1);
+
+    // .1 = 1.3V range / 13 datapoints on curve
```
Comment (mccauleyma) [inline shepherd_bms/src/analyzer.cpp:394]: This should be mentioned in the comments with the LUT. Important to know for setting the LUT/curve

File: shepherd_bms/src/compute.cpp [hunk 3]
```diff
@@ -282,3 +282,47 @@ void ComputeInterface::MCCallback(const CAN_message_t &msg)
 {
     return;
 }
+
+uint8_t ComputeInterface::calcChargerLEDState(uint8_t state_of_charge, AccumulatorData_t *bms_data)
```
Comment (mccauleyma) [inline shepherd_bms/src/compute.cpp:286]: As mentioned before rebase, I'd like to see SoC added to bms data and then just take the bms data for this function, not adding SoC as an extra param

**Review by mccauleyma — CHANGES_REQUESTED** [review]
Thank you for moving to updated branch. Far easier to review

File: shepherd_bms/src/compute.cpp (hunk 3 above)
Comment (dyldonahue) [reply shepherd_bms/src/compute.cpp:286]: yeah plan was to still do this before merge, nick just wanted to rebase first. If okay with @nwdepatie ill check this branch out and push these last few changes in the am

**Review by dyldonahue — APPROVED** [review]

File: shepherd_bms/src/compute.cpp [hunk 4]
```diff
@@ -17,7 +17,7 @@ void ComputeInterface::enableCharging(bool enableCharging)
     isChargingEnabled = enableCharging;
 }
 
-FaultStatus_t ComputeInterface::sendChargingMessage(uint16_t voltageToSet, uint16_t currentToSet)
+FaultStatus_t ComputeInterface::sendChargingMessage(uint16_t voltageToSet, uint16_t currentToSet, uint8_t state_of_charge, AccumulatorData_t *bms_data)
```
Comment (mccauleyma) [inline shepherd_bms/src/compute.cpp:20]:
```suggestion
FaultStatus_t ComputeInterface::sendChargingMessage(uint16_t voltageToSet, uint16_t currentToSet, AccumulatorData_t *bms_data)
```
I think we're good to remove this now right? (+ remove from header file and call to function)

**Review by mccauleyma — COMMENTED** [review]
Is passing SoC separately still required? Looks like it was removed but is still in the function call

File: shepherd_bms/src/compute.cpp (hunk 4 above)
Comment (dyldonahue) [reply shepherd_bms/src/compute.cpp:20]: whoops didnt realize we called that in the telemetry broadcast, yes that can be removed ill do now

**Review by mccauleyma — APPROVED** [review]
Thank you thank you

**Comment by mccauleyma** [comment]
Oop does it not build now lmao

File: shepherd_bms/src/main.cpp [hunk 5]
```diff
@@ -409,7 +409,7 @@ void shepherdMain()
 			// Send CAN message, but not too often
 			if (chargeMessageTimer.isTimerExpired()) 
 			{
-				compute.sendChargingMessage(MAX_CHARGE_VOLT * NUM_CELLS_PER_CHIP * NUM_CHIPS, analyzer.bmsdata->chargeLimit);
+				compute.sendChargingMessage(MAX_CHARGE_VOLT * NUM_CELLS_PER_CHIP * NUM_CHIPS, analyzer.bmsdata->chargeLimit,analyzer.bmsdata->soc , analyzer.bmsdata);
```
Comment (mccauleyma) [inline shepherd_bms/src/main.cpp:412]:
```suggestion
				compute.sendChargingMessage(MAX_CHARGE_VOLT * NUM_CELLS_PER_CHIP * NUM_CHIPS, analyzer.bmsdata->chargeLimit, analyzer.bmsdata);
```

**Comment by dyldonahue** [comment]
yeah yeah looks like this was fucky before, someone else got weird with passing in Accdata before doing soc fix. im addessing rn

**Comment by dyldonahue** [comment]
@mccauleyma fixed now, i removed both soc and current limit since there was no need to send that individually either. Going to merge if ur all good with things now

**Pull request merged**